### ISSUE TITLE
Please add more collaborators

### ISSUE THREAD

**Issue by aaronshaf** [balderdashy/waterline#862]
There seem to be quite a few outstanding pull requests, etc.

**Comment by dmarcelino** [comment]
The outstanding issues / pull requests is also something that also worries me... Currently, there are 'a lot' of people willing to contribute but it seems multiple PR's are not getting merged in. Not responding and/or creating a way to merge/share ideas being worked on discourages future contributions... Some issues linger many months (e.g. #497) without any resolution which contribute to frustrate the people involved.

I suspect the waterline team is quite busy with Sails (v0.11 was just released) but a solution should be found for waterline to keep evolving and improving. Maybe envolve people not associated with sails?

PS: this is not to say I'm dissatisfied with waterline, it's a great ORM and the team has done a great job. Thanks!

**Comment by randallmeeker** [comment]
I suggest sailsjs hack day! 1 day month where collaborators are actively online responding to pull requests and answering questions of contributors. Something organised so those of us can plan around.

**Comment by devinivy** [comment]
I quite agree, and would be willing to help.  The number of PRs is getting a bit out of hand, and several of them pertain to bugs that are genuinely critical.  Some of them also pertain to features that are extremely welcomed.  I get the feeling some people who could be contributors are using their own forks right now.  I think that this is a great ORM too, and I really appreciate all the effort that has gone into the project!  At the same time the sheer number of PRs and outstanding issues is concerning.

**Comment by dmarcelino** [comment]
@devinivy, when I started using waterline some of your comments helped me quite a bit. Namely the one about many-to-many through associations, which helped me implement through associations on my app, thanks!

It also makes me feel unsettled that there is no word if some sought after features will be implemented or when. To name a few:
* Transactions - #755
* Recursive populate - #256 / #308
* Many-to-many through associations release - #705
* Custom join table names - #497 / #725
* Composite indexes - #109

Some are more complex than others but it would be good to hear from the team about these and maybe see them being given milestones?

**Comment by particlebanana** [comment]
@randallmeeker @devinivy @dmarcelino welcome to the Waterline core team! The only thing I ask is that any major commits you run by me first - shoot me an email. All features need to work across all our adapters.

This last piece (cross adapter) is why we don't have Transactions, Recursive populations, and composite indexes yet. Of the list above custom join table names, m-m through and composite are the easiest ones to get in anytime soon.

**Comment by dmarcelino** [comment]
Thanks @particlebanana, I'll try to help where I can.

Regarding cross adapter features (Transactions, Recursive populations, and composite indexes), it would be great to know if there are any discussions or plans. I read some of the stuff on mikermcneil/waterline2 branch and understand he reached a sorte of roadblock. Are there any other efforts being taken?

**Comment by particlebanana** [comment]
ps know anyone who would be interested in helping maintain the mongo adapter?

**Comment by devinivy** [comment]
Wonderful– thanks very much.  I can certainly play by those rules :)
I've looked at the mongo adapter, but I don't use it actively.  That said, I could help monitor code changes/PRs around implementing deep associations, which I know some mongo users are thinking about.

**Comment by randallmeeker** [comment]
@particlebanana honored sir. I'd be willing to "help out" over at mongo. To be honest I am an SQL guy, but so far all my sails projects have been mongo and I'm starting to learn my way around the repo (first pull request today!)

Also I know that Waterline 2 is being worked on, is their a roadmap? Are some of these features already getting baked in, should we be looking at more bugs for current waterline and look at waterline 2 for features?

**Comment by tjwebb** [comment]
+1 @randallmeeker !

**Comment by devinivy** [comment]
Yea, I'm inclined to say that the bugs–especially the type that are kicking around right now–should really be first priority.  Recursive population would definitely fit more naturally into a new codebase, as would composite keys and and related adapter-API/schema updates, which themselves may require a breaking change like WL2 if not 0.11.0.

**Comment by dmarcelino** [comment]
I agree with bugs being the top priority. It would be good to go through the open issues and clean them up (prioritise / close duplicates / close issues that are no longer relevant / label accordingly / etc). PRs could also use some attention. Some of the issues/PRs are as old as 2013! :)

**Comment by devinivy** [comment]
To be fair, at least a few of the PRs are marked for the 0.11 milestone!

Can we close this issue and make separate issues for these conversations?  Waterline does have more collaborators now who are willing to help, so this seems to be successful!

**Comment by particlebanana** [comment]
Love the enthusiasm guys! One of you has to close it though :smile:

**Comment by tjwebb** [comment]
Thanks everyone for your concern and support. We'll continue to working to figure out the best way to manage things moving forward.

**Comment by dmarcelino** [comment]
Ooops, I'm a bit late... I was just gonna suggest one last thing: how do you guys feel about waterline having it's own gitter room? I use waterline standalone and as such it's hard for me to follow sails gitter room.

**Comment by mgenev** [comment]
@particlebanana 
I'm interested in helping with the mongo adapter

**Comment by particlebanana** [comment]
Thanks @mgenev just added you!

**Comment by dmarcelino** [comment]
Good stuff :+1:

**Comment by mgenev** [comment]
Thank you @particlebanana! Exciting.

I think the latest PR you merged into sails-mongo fixes a bug I ran into with date range queries. What is the procedure about pushing to NPM?

**Comment by devinivy** [comment]
Typically just email the lead maintainer (@particlebanana for WL).  It sounds like waterline will have an npm publish occur very soon.

**Comment by particlebanana** [comment]
@mgenev just published a new version

**Comment by mgenev** [comment]
@particlebanana awesome! I verified that date range queries now work

**Comment by dmarcelino** [comment]
@particlebanana, @rwoverdijk has also expressed that he would like to contribute to waterline (and blueprints). He already contributed to waterline-sequel.

**Comment by gaurav21r** [comment]
Maybe off topic, just for curiosity why isn't @mikermcneil active at all in Waterline? Wasn't he a **Founding Father** ? :smile: I've listened to his talks and he has some amazing ideas...

**Comment by randallmeeker** [comment]
he's busy installing lasers on squirrels, it has a very important theoretical bearing on waterline's fundamental development. 

but he was working on https://github.com/mikermcneil/waterline2 and then that stopped.

**Comment by tjwebb** [comment]
@randallmeeker FYI: I killed Waterline2 because there's No Good Reason for a top-to-bottom rewrite. I've decided to direct focus on strengthening the fundamentals of Waterline, and incorporate the WL2 features into the current adapter system. Moving forward with WL2 would dilute our efforts and fracture the ecosystem.

@gaurav21r day-to-day operations are still managed by [Balderdash](balderdash.co). @mikermcneil is working on a really cool downstream project called [Treeline](treeline.io).

**Comment by atiertant** [comment]
we love waterline ! it's a realy good project
we are trying to help the waterline project the most that we can.
i understand @mikermcneil is working on treeline, @dmarcelino and @devinivy are realy active,they do their best but it's true that PR are staying long time without any answer.

**Comment by devinivy** [comment]
I agree that it's a problem.  I go through periods where I can be more active in the project, and right now I've not been able to put as much time into waterline as in the recent past.  That said, I'm keeping up to date with the issues across all the repos on which I'm a collaborator.  The waterline ecosystem is a bit larger than the number of highly active collaborators at the moment, and the PRs are often of high complexity due to the nature of the tool.  But I do really want to see v0.11 come out, and I'll be making a push for that in the near future.  @atiertant your contributions have been paramount, and those that are pending (i.e. deep populate) will surely make their way into the project.

Perhaps we should create a repo for high-level discussions like this?

**Comment by dmarcelino** [comment]
Hey, I just came back from vacation and next week I'm starting a new job so I'll probably be less active. I'll still be following issues and contributing where I can and, same as @devinivy, I really want to see `v0.11` come out. If we can resolve all issues assigned to it will fix waterline's most common complaints.

@atiertant, thank you for your contributions, they are really moving the project forward!

**Comment by gaurav21r** [comment]
How about a suggestion? Why not pitch Waterline (not Sails or Treeline or any other connected product) to an Open Source Foundation or a Corporation for funding? Looking at these comments there are many hard core passionate lovers of Waterline but no full timers.

I recently heard of Microsoft expressing interest in collaborating on Sails. I guess for auto integrations into their Microsoft Azure PaaS.

Maybe the Apache Foundation or The Dojo Foundation can be approached?

**Comment by tjwebb** [comment]
> Why not pitch Waterline (not Sails or Treeline or any other connected product) to an Open Source Foundation or a Corporation for funding?

We tried that, and @mikermcneil backed out at the last minute. It's starting to look like he promised Sails.js to his Treeline investors.

**Comment by mikermcneil** [comment]
How is

**Comment by mikermcneil** [comment]
I'm getting tired of this.

I think it's important to point out three things here:
1. We're 100% committed to the eventual transfer of Sails and Waterline to an open-source foundation, but not until we reach 1.0 status.  
2. The _only_ contributor who was removed from this project was Travis Webb (@tjwebb).  This was due to him making repeated public attacks on my character and harassing my team.
3. I never made any kind of agreement with Travis Webb, and I never promised Sails.js to anyone; including Treeline's investors.

If you have any questions, please feel free to reach out to me privately.

**Comment by mikermcneil** [comment]
Sorry, had to lock this temporarily while permissuons were fixed up.

@particlebanana, @dmarcelino, @devinivy and everyone else contributing to Waterline have done a great job. This is a large project, with lots of moving pieces. New feature development will go through swings-- the most important thing is that we're able to address pending bugs in documented features, and holes in documentation and tests. I'm prepared to help with that as much as I can.

**Comment by devinivy** [comment]
Seems like it might be important to tighten up the docs, so we can decided if a bug is breaking a documented feature.  There are some well-defined issues in the waterline-docs repo that would be great to knock out.

**Comment by mikermcneil** [comment]
@devinivy I think that'd be a great idea.

**Comment by mikermcneil** [comment]
To bring some closure to this issue: the rest of the core team and I are as committed as ever to the continued development and support of Sails and Waterline.  If you have questions or concerns about our team, what we're doing, or where we're headed, please don't hesitate to [reach out to me directly](mission@treeline.io).

> For folks landing on this issue from search engines: please see https://github.com/balderdashy/sails/issues/3429#issuecomment-168801414 for some more background.
